src/backend/vulkan/src/lib.rs

```diff
@@ -343,29 +369,13 @@ impl core::Adapter<Backend> for Adapter {
             }
         }).collect::<Vec<_>>();
 
-        // Create associated command queues for each queue type
-        let queues = queue_infos.iter().flat_map(|info| {
-            (0..info.queue_count).map(|id| {
-                let queue_raw = unsafe {
-                    device.raw.0.get_device_queue(info.queue_family_index, id)
-                };
-                let queue = CommandQueue {
-                    raw: Arc::new(queue_raw),
-                    device: device.raw.clone(),
-                    family_index: info.queue_family_index,
-                };
-                unsafe {
-                    core::CommandQueue::new(queue)
-                }
-            }).collect::<Vec<_>>()
-        }).collect();
-
+        let device_arc = device.raw.clone();
         core::Gpu {
             device,
-            general_queues: queues,
-            graphics_queues: Vec::new(),
-            compute_queues: Vec::new(),
-            transfer_queues: Vec::new(),
+            general_queues: collect_queues(queue_descs, &device_arc, QueueType::General),
```

Comment:
@msiglreith 
it was also tempting to have queue capability trait with a method returning `QueueType` (aka our `Formatted::get_format()`), but I'd rather not make this complication if this initialization code is the only use case :)